dali/operators/math/expressions/expression_factory_instances/expression_impl_factory.h

```diff
@@ -105,6 +105,62 @@ std::unique_ptr<ExprImplBase> ExprImplFactoryBinOp(const ExprFunc &expr) {
   return result;
 }
 
+
+/**
+ * @brief Inspect `expr` to transform runtime information to static information, obtain the
+ *        implementation for ternary op (executor for given expression) and return it.
+ *
+ * The static type switch goes over the output type and input kinds.
+ * This is ternary case and seven possible input kind combinations are supported,
+ * that is all combinations of Tensors and Constants, except (Constant, Constant, Constant)
+ *
+ * @tparam Impl template that maps binary Arithmetic Op and input/output types
+ *                          to a functor that can execute it over a tile of three tensors.
+ *                          The implementation need to loop over inputs and output.
+ *                          @see ExprImplCpuTernary or ExprImplGpuTernary
+ * @param expr The expression node for which we return the executor
+ */
+template <ArithmeticOp op, typename Backend,
+          template <ArithmeticOp, typename, bool, bool, bool> class ImplsAll>
+std::unique_ptr<ExprImplBase> ExprImplFactoryTernaryOp2(const ExprFunc &expr) {
+  std::unique_ptr<ExprImplBase> result;
+  int scalar_count = 0;
+  bool is_tensor[3];
+  for (int i = 0; i < 3; i++) {
+    is_tensor[i] = !IsScalarLike(expr[i]);
+    if (!is_tensor[i]) {
```

Comment:
Double !?

Reply:
Only here, I need to check if I don't use three scalars as constants, as this is not a supported case.